### PR TITLE
Fix misspelled words and typos in docs + run a spell checker in CI

### PR DESCRIPTION
**Pull request by thedrow**
I had 15 minutes to spare. This is the result :)

### CODE REVIEW

**Comment by thedrow** [comment]
Seems like travis refuses to run node related stuff in CI.
We'll figure this out later :)

**Comment by dwhitena** [comment]
@sjezewski or @msteffen could you take a look here as this impacts Travis? I'm running the CI for this now, and will merge if that part looks ok to you.

**Comment by dwhitena** [comment]
Ping on this @sjezewski or @msteffen.

**Comment by thedrow** [comment]
Ping?

**Comment by sjezewski** [comment]
This is nifty.

The concern I have is that this adds a node install to CI. We're a bit sensitive to adding to CI build times lately. I guess when we get around to optimizing build times, we can cache the node download.

Idk. What do you think @jdoliner ?

**Comment by jdoliner** [comment]
I'd be interested to see what the build times are with this patch. But my general feeling is that adding a node dependency to CI to get a spellchecker doesn't quite work out from a cost benefit perspective.

**Comment by sjezewski** [comment]
Yea ... looking into it ... (travis is nice and annotates each command's elapsed time) it took the `before_install.sh` from ~1s to ~20s.

Not as bad as I would've thought. But not good either.

**Comment by sjezewski** [comment]
So we discussed this, and are good w the actual spell check changes, and even the new make task, but don't want it running on CI. Can you rip out that part?

**Comment by thedrow** [comment]
Sure. I can also try to activate one of the pre-installed nodejs programs somehow. I can separate it to a second PR.

**Comment by thedrow** [comment]
@sjezewski done :)

File: etc/testing/ci/before_install.sh [hunk 1]
```diff
@@ -20,4 +20,3 @@ curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/v1.8.
 curl -Lo minikube https://storage.googleapis.com/minikube/releases/v0.23.0/minikube-linux-amd64 && \
     chmod +x ./minikube && \
     sudo mv ./minikube /usr/local/bin/minikube
-
```
Comment (sjezewski) [inline etc/testing/ci/before_install.sh]: Do you mind putting this line back? Just makes the diff a bit nicer.

**Comment by thedrow** [comment]
@sjezewski Ready for merge :)

**Comment by thedrow** [comment]
@dwhitena Ping?

**Comment by dwhitena** [comment]
@thedrow @sjezewski new CI run here: https://github.com/pachyderm/pachyderm/pull/2579

**Comment by dwhitena** [comment]
@sjezewski CI passed. Is this ready to merge?

**Comment by sjezewski** [comment]
I mean, if you look at the diff, the line I mentioned was added back, but w different whitespace, so now it's a different whitespace difference.

But since this PR process incurs so much overhead (we have to build the PR, etc) and its just whitespace, yea lets go ahead and just merge it.

**Pull request merged**